pkg/ottl/ottlfuncs/func_replace_all_matches.go

```diff
@@ -69,6 +71,17 @@ func replaceAllMatches[K any](target ottl.PMapGetter[K], pattern string, replace
 				return nil, fmt.Errorf("replacement value is not a string")
 			}
 			replacementVal = replacementValStr
+			if !replacementFormat.IsEmpty() { // If replacementFormat is not empty, add it to the replacement value
+				formatString := replacementFormat.Get()
+				formatStringVal, err_fmt := formatString.Get(ctx, tCtx)
+				if err_fmt != nil {
+					return "", err_fmt
+				}
+				if strings.Count(formatStringVal, "%s") > 1 || strings.Count(formatStringVal, "%") < 1 {
```

Comment:
I think we need to update the logic here, I can think of a few cases we need to make sure don't pass:

- `%d`
- `%s %d`
- `%d %d`

We should probably check that `%s` is used exactly once, and that there's no presence of `%(?!s)` in the string.

Reply:
I've updated the logic to use regex for this
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30837/commits/9599cf981f28daafc2aa9e6891f488d29e8ced9c